### DISCUSSION TITLE
Help，when use the redis client with default options, will it manage the connection pool ?

### DISCUSSION THREAD

**Discussion by hongfangao** [redis/go-redis#1919] [Q&A]
I'm a new man for the go-redis. Now I wonder that if the redis client manage the connection pool automatically?

```
// global variable
var myRedis *redis.Client
// init redis
init() {
    ...
}
// use it
myRedis.Get("key")
```
What means "manage automatically" ? For example, I mean that: 
As the code, if the sentence "myRedis.Get("key")" is done by a lot of goroutine, will the redis client automatically create some new connections to get a better performance ?
And when the sentence is not done frequently, will the redis client close the redundant connections automatically ?
Thanks for your help !

**Comment by rambabuwhy** [comment]
same here. need clarification.  any expert here ?

**Comment by lubaoyilang** [comment]
I also want to know

Looking at the source code, Pipeline's Exec() does not call close
but,
Pipelined(ctx context.Context, fn func(Pipeliner) error)
close is called automatically

**Reply by baylac** [reply]
I think close is not needed, after Execute() the pipe can be reused.

**Comment by vmihailenco** [comment]
go-redis automatically manages connections - it should open new connections when needed and also close idle connections. See Options.PoolSize, Options.IdleTimeout, and Options.MaxConnAge.

>Looking at the source code, Pipeline's Exec() does not call close

You should close the pipeline when it is not needed. `Exec` does not close the pipeline.
